### PR TITLE
Conditionally restart NGINX

### PR DESCRIPTION
**Pull request by fomojola**
Before restarting nginx, check if the current app is using a proxy and if the selected proxy is in fact NGINX. Only then should we bother invoking the nginx reload.

Fixes https://github.com/dokku/dokku/issues/4987

### CODE REVIEW

File: plugins/nginx-vhosts/functions [hunk 1]
```diff
@@ -128,7 +128,13 @@ validate_nginx() {
 
 restart_nginx() {
   declare desc="restart nginx for given distros"
-  fn-nginx-vhosts-nginx-init-cmd "reload"
+  local PROXY_ENABLED="$(plugn trigger proxy-is-enabled "$APP")" 
+  local PROXY_TYPE="$(plugn trigger proxy-type "$APP")"      
+  if [[ "$PROXY_ENABLED" == "true" ]]; then                  
+    if [[ "$PROXY_TYPE" == "nginx" ]]; then
```
Comment (josegonzalez) [inline plugins/nginx-vhosts/functions:134]: Mind changing this to:

```
if [[ "$PROXY_ENABLED" == "true" ]] && [[ "$PROXY_TYPE" == "nginx" ]]; then
```

Also, there is a formatting issue, should be easy to check CI output to see whats up :)

**Review by josegonzalez — CHANGES_REQUESTED** [review]

**Comment by josegonzalez** [comment]
Thanks for the pull request, it's much appreciated!

File: plugins/nginx-vhosts/functions (hunk 1 above)
Comment (fomojola) [reply plugins/nginx-vhosts/functions]: And done (I think)! I don't spend enough time in shell land: I wasn't confident I could stick the double.

Comment (josegonzalez) [reply plugins/nginx-vhosts/functions]: No worries, you'll get more comfortable as you make more pull requests :)

**Comment by fomojola** [comment]
Had to install shfmt: should be good to go now.

**Pull request merged**